mopidy/audio/tags.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import absolute_import, unicode_literals
+
+import collections
+import datetime
+import logging
+import numbers
+
+import gi
+gi.require_version('Gst', '1.0')
+from gi.repository import GLib, Gst
+Gst.is_initialized() or Gst.init()
+
+from mopidy import compat
+from mopidy.internal import log
+from mopidy.models import Album, Artist, Track
+
+
+logger = logging.getLogger(__name__)
+
+
+def convert_taglist(taglist):
+    """Convert a :class:`Gst.TagList` to plain Python types.
+
+    Knows how to convert:
+
+    - Dates
+    - Buffers
+    - Numbers
+    - Strings
+    - Booleans
+
+    Unknown types will be ignored and trace logged. Tag keys are all strings
+    defined as part GStreamer under GstTagList_.
+
+    .. _GstTagList: https://developer.gnome.org/gstreamer/stable/\
+gstreamer-GstTagList.html
+
+    :param taglist: A GStreamer taglist to be converted.
+    :type taglist: :class:`Gst.TagList`
+    :rtype: dictionary of tag keys with a list of values.
+    """
+    result = collections.defaultdict(list)
+
+    for n in range(taglist.n_tags()):
+        tag = taglist.nth_tag_name(n)
+
+        for i in range(taglist.get_tag_size(tag)):
+            value = taglist.get_value_index(tag, i)
+
+            if isinstance(value, GLib.Date):
```

Comment:
We convert it to an ISO-8601 string in both cases, just with the help of `datetime.date` in the `GLib.Date` case and with the help of GStreamer in the `Gst.DateTime` case.